python_modules/libraries/dagster-airflow/dagster_airflow_tests/test_dagster_job_factory/test_load_assets_airflow_2.py

```diff
@@ -68,6 +64,17 @@ def new_upstream_asset():
             },
         )
 
+        resources = None
+
+        if assets:
+            first_asset = next(iter(assets))
+            if "io_manager" in first_asset.resource_defs:
+                resources = {"io_manager": first_asset.resource_defs["io_manager"]}
+
+        @asset(resource_defs=resources)
+        def new_upstream_asset():
+            return 1
+
```

Comment:
Funny enough this appeared to have found a real issue in the dagster-airflow work where we could get in state of providing conflicting resource definitions. This code is structured a bit oddly, where the job is created in load_assets_from_airflow_dag, but then discarded after it's contents have been shuffled into a single graph-backed AssetsDefinition.

@Ramshackle-Jamathon, @sryza you two might want to talk about the best approach here.